src/main/java/com/ericsson/ei/notifications/HttpRequest.java

```diff
@@ -154,45 +156,32 @@ private void addAuthenticationData() throws AuthenticationException {
         String username = subscriptionField.get("userName");
         String password = subscriptionField.get("password");
 
-        boolean authenticationDetailsProvided = isAuthenticationDetailsProvided(authType, username,
-                password);
+        boolean authenticationDetailsProvided = EncryptionFormatter.verifyAuthenticationDetails(
+                authType, username, password);
         if (!authenticationDetailsProvided) {
             return;
         }
+        if (encryptor != null && encryptor.isJasyptPasswordSet()
+                && EncryptionFormatter.isEncrypted(password)) {
+            password = doDecryption(password);
+        }
 
         String encoding = Base64.getEncoder()
                                 .encodeToString((username + ":" + password).getBytes());
         this.headers.add("Authorization", "Basic " + encoding);
         LOGGER.debug("Successfully added header for 'Authorization'");
 
-        if (authType.equals(AUTHENTICATION_TYPE_BASIC_AUTH_JENKINS_CSRF)) {
+        if (authType.equals(AuthenticationType.BASIC_AUTH_JENKINS_CSRF.getValue())) {
             JsonNode crumb = jenkinsCrumb.fetchJenkinsCrumb(encoding, this.url);
             addJenkinsCrumbData(crumb);
         }
 
     }
 
-    /**
-     * Returns a boolean indicating that authentication details was provided in
-     * the subscription.
-     *
-     * @param authType
-     * @param username
-     * @param password
-     * @return
-     */
-    private boolean isAuthenticationDetailsProvided(String authType, String username,
-            String password) {
-        if (authType.isEmpty() || authType.equals(AUTHENTICATION_TYPE_NO_AUTH)) {
-            return false;
-        }
-
-        if (username.isEmpty() && password.isEmpty()) {
-            LOGGER.error("userName/password field in subscription is missing.");
-            return false;
-        }
-
-        return true;
+    private String doDecryption(String password) {
+        String encryptedPassword = EncryptionFormatter.removeEncryptionParentheses(password);
+        password = encryptor.decrypt(encryptedPassword);
```

Comment:
maybe encryptor.decrypt should contain a check if password has EncryptionParentheses and remove them, thus this does not have to be done everywhere decryption is done.

Reply:
Matias suggested the same. I could add it to decrypt but then it's coupled to EncryptionFormatter and it does something more than just a simple task. Should the method name still remain the same even though it does something hidden?

Reply:
Yes, it would basicly prepare value for decryption and it could be a part of the decryption.

Reply:
Alright I'll change this. What I don't like with this solution however is that now Encryption cannot exist with EncryptionFormatter and also it will not decrypt a string unless it has the tag. So it's no longer a "simple" wrapper for a decryption library.